VideoPress: add core/embed transform from/to video block

retrofox opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes #

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* VideoPress: add core/embed transform from/to video block

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to block editor app
* Add an embed block
* Paste a VideoPress video URL. eg: `https://videopress.com/v/PupnE9f3`
* Select keep using the Embed block

<img width="649" alt="image" src="https://user-images.githubusercontent.com/77539/208120026-36801314-b609-4b39-bc33-ac98d5c18639.png">

* Confirm the block provides a transform to the VideoPress video block
<img width="661" alt="image" src="https://user-images.githubusercontent.com/77539/208124647-6772026a-3545-4cff-bc20-fee55decaeaf.png">

* Confirms it transforms properly
* Confirm now the new VideoPress video blocks provide a transform to the core Embed block too.

<img width="669" alt="image" src="https://user-images.githubusercontent.com/77539/208124851-2b793ffc-1ff3-4302-9acc-345c53cafcae.png">

retrofox committed:
[not verified] add core/embed to/from transform

retrofox committed:
[not verified] changelog

retrofox committed:
[not verified] fix isMatch from core/embed block

lhkowalski commented:
Testing on a JN site, I can confirm that the options to transform the blocks are present there, and the transformation happens as expected.

From `embed` to `VideoPress`:

<img width="600" alt="Screen Shot 2022-12-19 at 16 44 09" src="https://user-images.githubusercontent.com/6760046/208507413-9db2eef4-d718-4048-bd25-19bb1d6e13f6.png">

From `VideoPress` to `embed`:

<img width="600" alt="Screen Shot 2022-12-19 at 16 44 28" src="https://user-images.githubusercontent.com/6760046/208507524-49be43aa-6524-473b-92f0-03ea798c8ec8.png">

retrofox committed:
revert exclugind src/client files
Revert the change introduced here https://github.com/Automattic/jetpack/pull/27926

dhasilva reviewed (commented):
Working on simple site as well after the last commit
![2022-12-20_09-56-35](https://user-images.githubusercontent.com/8486249/208672814-13d7593e-f326-4f2c-9be3-200fce55e0f6.png)
![2022-12-20_09-55-46](https://user-images.githubusercontent.com/8486249/208672824-9b42a365-2b67-41b8-ae5c-d52d53ef72f0.png)

dhasilva reviewed (approved):
LGTM
